### PR TITLE
configurable interval for syncing permissions

### PR DESCRIPTION
**Pull request by mollylogue**

### CODE REVIEW

**Review by efritz — APPROVED** [review]
Can you ensure that this branch passes the tests on a `main-dry-run/*` branch?

**Comment by efritz** [comment]
@mollylogue These look like insights visual failures only; I just wanted to make sure all of the pods started successfully (not deadlocked on waiting for the frontend config).

This build looks 👍 : https://buildkite.com/sourcegraph/sourcegraph/builds/129689

**Comment by mollylogue** [comment]
> 

Random visual failures make me v nervous 😅

File: enterprise/internal/authz/authz.go [hunk 1]
```diff
@@ -252,3 +253,10 @@ func ProviderFromExternalService(siteConfig schema.SiteConfiguration, svc *types
 	}
 	return providers[0], nil
 }
+
+func ParseInterval(interval int) time.Duration {
+	if interval == 0 {
```
Comment (unknwon) [inline enterprise/internal/authz/authz.go:258]:
```suggestion
	if interval <= 0 {
```

nit: just to be very cautious ...

File: enterprise/internal/authz/authz.go [hunk 2]
```diff
@@ -252,3 +253,10 @@ func ProviderFromExternalService(siteConfig schema.SiteConfiguration, svc *types
 	}
 	return providers[0], nil
 }
+
+func ParseInterval(interval int) time.Duration {
```
Comment (unknwon) [inline enterprise/internal/authz/authz.go:257]: Based on changes in this PR, any reason we don't call `conf.Get().AuthzRefreshInterval` directly within this function and renamed it to `AuthzRefreshInterval()`?

File: schema/site.schema.json [hunk 3]
```diff
@@ -700,6 +700,11 @@
       "type": "boolean",
       "default": false
     },
+    "authz.refreshInterval": {
+      "description": "Time interval (in seconds) between requests to refresh oauth token scopes",
```
Comment (unknwon) [inline schema/site.schema.json:704]:
```suggestion
      "description": "Time interval (in seconds) of how often each component picks up authorization changes in external services.",
```

**Review by unknwon — APPROVED** [review]
Approve to unblock, left comments in previous review.

**Review by indradhanush — APPROVED** [review]
Looks good. Approve to unblock. Agree with Joe's comment about `AuthzRefreshInterval()`.

File: enterprise/internal/authz/authz.go (hunk 2 above)
Comment (mollylogue) [reply enterprise/internal/authz/authz.go:257]: So, mostly I was following the convention that I've been seeing where we get these config variables at the highest level possible, which is why this is just a helper method to parse the interval. But if we're cool with this pulling in the interval from the config, I'm happy to change. It makes this way cleaner.

**Pull request merged**